### PR TITLE
qemu.test: MigrationMultiHost

### PR DESCRIPTION
**Pull request by XueqiangWei**
super class for multihost migration, include some public functions

Signed-off-by: Xueqiang Wei <xuwei@redhat.com>

### CODE REVIEW

**Comment by XueqiangWei** [comment]
ID 1342353

File: qemu/tests/MigrationMultiHost.py [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+from generic.tests import kdump
+
+
+class MigrationBase(object):
+
+    """Class that provides some general functions for multi-host migration."""
+
+    def __setup__(self, test, params, env, srchost, dsthost):
+
+        """initialize some public params
```
Comment (CongLi) [inline qemu/tests/MigrationMultiHost.py]: 2 lines ?

Comment (XueqiangWei) [inline qemu/tests/MigrationMultiHost.py]: yes, it is suggestion in pep8
$pep8 MigrationMultiHost.py 
$

File: qemu/tests/MigrationMultiHost.py [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+                                              "pkill -9 ping")
+        self.migration_timeout = int(params.get("migration_timeout",
+                                                "1500"))
+        self.login_timeout = 480
+        self.stop_migrate = False
+        self.migrate_count = int(params.get("migrate_count", 1))
+        self.id = {"src": self.srchost,
+                   "dst": self.dsthost,
+                   "type": "file_transfer"}
+        self.capabilitys = params.get("capabilitys", "").split()
```
Comment (XueqiangWei) [inline qemu/tests/MigrationMultiHost.py]: params.object("capabilitys")

File: qemu/tests/MigrationMultiHost.py [hunk 3]
```diff
@@ -0,0 +1,406 @@
...
+        self.capabilitys_state = params.get("capabilitys_state", "").split()
```
Comment (XueqiangWei) [inline qemu/tests/MigrationMultiHost.py]: params.object("capabilitys_state")

File: qemu/tests/MigrationMultiHost.py [hunk 4]
```diff
@@ -0,0 +1,406 @@
...
+        for i in range(0, len(self.capabilitys_state)):
+            if self.capabilitys_state[i].strip() == "enable":
+                self.capabilitys_state[i] = True
+            else:
+                self.capabilitys_state[i] = False
+        self.cache_size = params.get("cache_size", "").split()
```
Comment (XueqiangWei) [inline qemu/tests/MigrationMultiHost.py]: params.object("cache_size")

**Comment by vivianQizhu** [comment]
I think we should include migrate_set_downtime, migrate_cancel, migrate_set_parameter here.

**Comment by XueqiangWei** [comment]
@vivianQizhu migrate_set_downtime and migrate_cancel are added, migrate_set_parameter will be added when develop related case. Please help review, many thanks.

File: qemu/tests/MigrationMultiHost.py [hunk 5]
```diff
@@ -0,0 +1,464 @@
...
+        :param nvcpu: which is used to trigger a crash
+        :param crash_cmd: which is triggered crash command
+        """
+
+        vm = mig_data.vms[0]
+        kdump.preprocess_kdump(vm, login_timeout)
+        kdump.kdump_enable(vm, vm.name, crash_kernel_prob_cmd,
+                           kernel_param_cmd, kdump_enable_cmd,
+                           login_timeout)
+        error.context("Kdump Testing, force the Linux kernel to crash",
```
Comment (suqinhuang) [inline qemu/tests/MigrationMultiHost.py]: need to setup kdump for all cases, right?

**Comment by suqinhuang** [comment]
did you make some mistake? I found the branch is migration_multi_host_with_xbzrle

File: qemu/tests/MigrationMultiHost.py (hunk 5 above)
Comment (XueqiangWei) [inline qemu/tests/MigrationMultiHost.py]: No, this is just a public method.

**Comment by XueqiangWei** [comment]
this file is a super class for multi-host migration, define some public functions.

**Comment by XueqiangWei** [comment]
add this pullreq to framework, so close it. And refer to #727

**Pull request closed without merging**